CT-1572 copy latest version of repo to docker image when building

misaka opened the pull request:
Apparently our base image has the context automatically copied, while subsequent
images we create from it do not, so we have to add an explicit COPY command to
ensure the code is up-to-date in the new image we build with Dockerfile

EdwardAndress commented on app/models/case.rb:
```diff
@@ -150,6 +150,14 @@ def attachments_dir(attachment_type)
     "#{Digest::SHA1.hexdigest(id.to_s)}/#{attachment_type}"
   end
 
+  def whose_it_with
```
Should this be `who_its_with`?

aliuk2012 replied:
👍

EdwardAndress commented on app/views/cases/index.html.slim:
```diff
@@ -26,6 +27,8 @@
         th scope='col'
           = t('.status')
+        th scope='col'
+          = "Whose it with"
```
Does this need to be updated too?

aliuk2012 replied:
(Doh) Thats should have been in locales. Well spotted

misaka committed:
CT-1572 copy latest version of repo to docker image when building

Apparently our base image has the context automatically copied, while subsequent
images we create from it do not, so we have to add an explicit COPY command to
ensure the code is up-to-date in the new image we build with Dockerfile

misaka committed:
CT-1572 remove existing app directory when building base

Files were being left over from the base image when copying the new image
overtop of it. As the base image doesn't require the app at all, it's safer to
just remove all the app files.

misaka committed:
CT-1572 tidying up of Docker env: add WORKDIR, -y to apt-get update

misaka committed:
CT-1572 add comment with insights into why we rm in Dockerfile.base
